client/landing/login/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Global polyfills
+ */
+import 'boot/polyfills';
+import { hydrate, render } from 'controller/web-util';
+
+/**
+ * External dependencies
+ */
+import page from 'page';
+/**
+ * Internal dependencies
+ */
+import createStore from './store';
+import { setupMiddlewares } from './common';
+import initLoginSection from 'login';
+
+// goofy import for environment badge, which is SSR'd
+import 'components/environment-badge/style.scss';
+
+// Create Redux store
+const store = createStore();
+
+setupMiddlewares( store );
+
+page( '*', ( context, next ) => {
+	context.store = store;
+	next();
+} );
+
+page.exit( '*', ( context, next ) => {
+	context.store = store;
+	next();
+} );
+
+initLoginSection( page );
+
+page( '*', ( context, next ) => {
+	if ( context.serverSideRender ) {
+		hydrate( context );
+	} else {
+		render( context );
+	}
+	next();
+} );
```

Comment:
Instead of this special handler, just pass an enhanced router to `initLoginSection`:
```js
initLoginSection( ( route, ...handlers ) => page( route, ...handlers, renderHandler ) );
function renderHandler( context, next ) {
  ( context.serverSideRender ? hydrate : render )( context );
  next();
}
```

Context: the section init function is passed a `router` object as argument, and inside it, you can setup route handlers either with `router` or `page`:
```js
export default router => {
  page( '/x', showX, makeLayout, clientRender );
  router( '/y', showY, makeLayout );
}
```
Note that `clientRender` is missing in the `router` call. That's because it's added automatically at the end, and it's Express.js or page.js function, depending on where the code runs.

The `router` thing is intended for isomorphic pages. See also the [`clientRouter`](https://github.com/Automattic/wp-calypso/blob/7aaf16531bb4d948e6b49b6df14e19e580354650/client/controller/index.web.js#L61-L63) function and its [usage in the section loader](https://github.com/Automattic/wp-calypso/blob/7aaf16531bb4d948e6b49b6df14e19e580354650/client/sections-middleware.js#L40).

Reply:
Done! Thanks for the suggestion!